### ISSUE TITLE
More convenient way to authorize a new device

### ISSUE THREAD

**Issue by koyuawsmbrtn** [element-hq/element-meta#1496]
**Is your suggestion related to a problem? Please describe.**

It's kinda inconvenient when I reset my PC a couple of times a year and I have to start from scratch and I have to get through like a thousand prompts just to sign me in and import my keys and sometimes some keys don't get imported and I have to grab my phone again to do the rest.

**Describe the solution you'd like**

That's where I wasn't sure where to put this in. Wouldn't it be cooler when the user just had to scan a QR code and then they're signed in with all keys imported and ready to back new ones up?

**Describe alternatives you've considered**

This seems to be the most convenient solution so far.

**Additional context**

In case you don't have your phone with you, you could just go and generate a paperkey that can be used for authentication like Keybase does it.

![image](https://user-images.githubusercontent.com/40538332/70388326-7b317a00-19b0-11ea-8e19-e27a2a545c21.png)

**Comment by aaronraimist** [comment]
Related: vector-im/element-meta#1561

**Comment by MadLittleMods** [comment]
Verifying your Element Android device via QR code works nowadays. AFAICT, it's not possible on iOS yet. In the iOS simulator, it only gave me an option to verify by emoji.

Element Android:

<img src="https://user-images.githubusercontent.com/558581/164528556-abb69018-9d42-4460-b43a-7fe80da9e7a3.png" width="450">


Element Web/Desktop:

![](https://user-images.githubusercontent.com/558581/164528684-e30e14e5-dfcf-45f9-86aa-fbe413014333.png)

**Comment by koyuawsmbrtn** [comment]
Yeah, this is device verification, but the whole sign in process with the keys already imported and devices verified is not a thing yet. I mean if your phone can already verify your session then it should also be able to sign you in. The best UX I have seen for this process would be the QR code login for Discord.

**Comment by t3chguy** [comment]
Login requires the password, which Element apps do not store, so a new spec change to request a one-time login code which could be used for this would be needed

**Comment by koyuawsmbrtn** [comment]
Exactly.

**Comment by richvdh** [comment]
I believe QR code login exists nowadays.
